test_conformance/spirv_new/spirv_asm/op_spec_constant_char_simple.spvasm32

```diff
@@ -0,0 +1,69 @@
+          %1 = OpExtInstImport "OpenCL.std"
+               OpMemoryModel Physical32 OpenCL
+               OpEntryPoint Kernel %10 "spec_const_kernel" %__spirv_BuiltInGlobalInvocationId
+         %39 = OpString "kernel_arg_type.spec_const_kernel.char*,char*,"
+               OpSource OpenCL_C 102000
+               OpDecorate %40 Alignment 4
+         %40 = OpDecorationGroup
+               OpDecorate %__spirv_BuiltInGlobalInvocationId BuiltIn GlobalInvocationId
+               OpDecorate %__spirv_BuiltInGlobalInvocationId Constant
+               OpDecorate %__spirv_BuiltInGlobalInvocationId LinkageAttributes "__spirv_BuiltInGlobalInvocationId" Import
+               OpDecorate %uchar_0 SpecId 101
+               OpDecorate %add NoSignedWrap
+               OpGroupDecorate %40 %numbers_buffer_addr %spec_constant_value_addr %id
+       %uint = OpTypeInt 32 0
+      %uchar = OpTypeInt 8 0
+    %uchar_0 = OpSpecConstant %uchar 0
+     %v3uint = OpTypeVector %uint 3
+%_ptr_Input_v3uint = OpTypePointer Input %v3uint
+       %void = OpTypeVoid
+%_ptr_CrossWorkgroup_uchar = OpTypePointer CrossWorkgroup %uchar
+          %9 = OpTypeFunction %void %_ptr_CrossWorkgroup_uchar %_ptr_CrossWorkgroup_uchar
+%_ptr_Function__ptr_CrossWorkgroup_uchar = OpTypePointer Function %_ptr_CrossWorkgroup_uchar
+%_ptr_Function_uint = OpTypePointer Function %uint
+%_ptr_Function_uchar = OpTypePointer Function %uchar
+%__spirv_BuiltInGlobalInvocationId = OpVariable %_ptr_Input_v3uint Input
+         %10 = OpFunction %void None %9
+%numbers_buffer = OpFunctionParameter %_ptr_CrossWorkgroup_uchar
+%spec_constant_value = OpFunctionParameter %_ptr_CrossWorkgroup_uchar
+      %entry = OpLabel
+%numbers_buffer_addr = OpVariable %_ptr_Function__ptr_CrossWorkgroup_uchar Function
+%spec_constant_value_addr = OpVariable %_ptr_Function__ptr_CrossWorkgroup_uchar Function
+         %id = OpVariable %_ptr_Function_uint Function
+               OpStore %numbers_buffer_addr %numbers_buffer Aligned 4
+               OpStore %spec_constant_value_addr %spec_constant_value Aligned 4
+         %20 = OpBitcast %_ptr_Function_uchar %id
+               OpLifetimeStart %20 4
+         %21 = OpLoad %v3uint %__spirv_BuiltInGlobalInvocationId
+       %call = OpCompositeExtract %uint %21 0
+               OpStore %id %call Aligned 4
+         %23 = OpLoad %_ptr_CrossWorkgroup_uchar %spec_constant_value_addr Aligned 4
+               OpStore %23 %uchar_0 Aligned 1
+         %25 = OpLoad %_ptr_CrossWorkgroup_uchar %numbers_buffer_addr Aligned 4
+         %26 = OpLoad %uint %id Aligned 4
+   %arrayidx = OpInBoundsPtrAccessChain %_ptr_CrossWorkgroup_uchar %25 %26
+         %28 = OpLoad %uchar %arrayidx Aligned 1
+       %conv = OpSConvert %uint %28
+         %30 = OpLoad %_ptr_CrossWorkgroup_uchar %spec_constant_value_addr Aligned 4
+         %31 = OpLoad %uchar %30 Aligned 1
+      %conv1 = OpSConvert %uint %31
+        %add = OpIAdd %uint %conv %conv1
+      %conv2 = OpUConvert %uchar %add
```

Comment:
This could be simplified by removing the `OpSConvert` and `OpUConvert` instructions.